src/mumble/PluginManager.cpp

```diff
@@ -146,8 +147,8 @@ bool PluginManager::selectActivePositionalDataPlugin() {
 	QMultiMap<QString, unsigned long long int> pidMap;
 	getProgramPIDs(pidMap);
 
-	const char* names[pidMap.size()];
-	uint64_t pids[pidMap.size()];
+	QScopedArrayPointer<const char*> names(new const char*[pidMap.size()]);
```

Comment:
Why not just QVector of strings?

Reply:
Because the API requires an array - or can QVector be converted to an array? :thinking_man:

Reply:
Just looked into the docs of QVector and it is indeed convertable. Though not quite sure if that is better as I don't need dynamic resizing. So in my eyes nothing is would be more performant than allocating an array of exactly that size - could be easily mistaken though...

Reply:
From https://doc.qt.io/qt-5/qvector.html#QVector-1:
> ### QVector::QVector(int size)
> Constructs a vector with an initial size of _size_ elements.
>
> The elements are initialized with a [default-constructed value](https://doc.qt.io/qt-5/containers.html#default-constructed-value).
>
> **See also** [resize](https://doc.qt.io/qt-5/qvector.html#resize)().
---
From https://doc.qt.io/qt-5/qvector.html#QVector-2:
> ### QVector::QVector(int size, const T &value)
> Constructs a vector with an initial size of _size_ elements. Each element is initialized with _value_.
>
> **See also** [resize](https://doc.qt.io/qt-5/qvector.html#resize)() and [fill](https://doc.qt.io/qt-5/qvector.html#fill)().

Reply:
@davidebeatrici That constructor is quite cumbersome to use for this case. I don't need to initialize all elements of the vector just so I can overwrite them with my actual values. Plus the only advantage I see in using a `QVector` over a plain array is the ability to use the `append` function allowing me to write the data-assignment in a simple forach-loop without having to carry an index with me. This isn't possible when using that constructor though as the list will actually be _filled_ by the respective amount of items so `append` won't overwrite them resulting in a vector twice as big - the first half being useless chunks.

I also thought about passing in two `QVector` instead of the `QMultiHash` in order to avoid re-copying the respective values in the first place but I can't seem to find a way that would let me store the `const char*` for the names in a way that guarantees the validity of those pointers once I exit `getProgramIDs` to actually use them (I assume that the `QString` object will free them as soon as they go out of scope - which they will unless I store them somewhere).

Reply:
Vector should have a data() member that returns pointer to the raw array.
I mainly think its easier to read, it doesn't look so ugly.

`QScopedArrayPointer<const char*> names(new const char*[pidMap.size()]);`
vs 
`QVector<const char*> names(pidMap.size())`

Also ideally you'd use pushBack and reserve, instead of setting size and then overwriting elements by index.

Reply:
> QVector<const char*> names(pidMap.size())

That'd have to be
```
QVector<const char*> names();
names.reserve(pidMap.size());
```
in order to 
> use pushBack and reserve, instead of setting size and then overwriting elements by index.


But good call on the `reserve`-function. Didn't spot that one :+1: